research/picai/prepare_data.py

```diff
@@ -0,0 +1,209 @@
+import argparse
+import json
+import os
+from pathlib import Path
+from typing import Optional, Sequence, Tuple
+
+import numpy as np
+import SimpleITK as sitk
+from preprocessing import (
+    AlignOriginAndDirection,
+    BinarizeAnnotation,
+    CentreCropAndOrPad,
+    PicaiCase,
+    PreprocessingSettings,
+    Resample,
+    ResampleToFirstScan,
+    preprocess,
+)
+
+DEFAULT_TRANSFORMS = [
+    Resample(),
+    CentreCropAndOrPad(),
+    ResampleToFirstScan(),
+    AlignOriginAndDirection(),
+    BinarizeAnnotation(),
+]
+
+
+def get_labels(paths_for_each_sample: Sequence[Tuple[Sequence[Path], Path]]) -> Sequence[float]:
+    _, annotation_paths = zip(*paths_for_each_sample)
+    case_labels = []
+    for path in annotation_paths:
+        annotation = sitk.ReadImage(path)
+        annotation_array = sitk.GetArrayFromImage(annotation)
+        case_label = float(np.max(annotation_array))
```

Comment:
Maybe a dumb question, but is this equivalent to `BinarizeAnnotation`?

Reply:
I will add a comment because this is definitely not clear. But the case label is simply whether or not the sample contains any foreground objects. If all of the entries are 0 than this will be 0, if there are any tumors, the result will be 1. So we can keep track of how many positive and negative samples we have which is helpful for setting focal loss